Bump MySQL version

Currently, the default MySQL version is 5.7.26. The latest release right now is 5.7.29.

Doing so backup restoring with `xtrabackup` fails. MySQL server fails to start with the following error:
```
2020-03-26T15:49:41.839523Z 0 [ERROR] mysqld: Table './mysql/user' is marked as crashed and should be repaired
2020-03-26T15:49:41.839561Z 0 [ERROR] Fatal error: Can't open and lock privilege tables: Table './mysql/user' is marked as crashed and should be repaired
2020-03-26T15:49:41.839576Z 0 [ERROR] Fatal error: Failed to initialize ACL/grant/time zones structures or failed to remove temporary table files.
2020-03-26T15:49:41.839667Z 0 [ERROR] Aborting
```

I will revert the default version of MySQL to `5.7.24`.

Using MySQL `5.7.24` those errors keep showing but are resolved by MySQL:

```
2020-03-26T15:52:22.524837Z 0 [Note] Server socket created on IP: '::'.
2020-03-26T15:52:22.559453Z 0 [ERROR] mysqld: Table './mysql/user' is marked as crashed and should be repaired
2020-03-26T15:52:22.559766Z 0 [Warning] Checking table:   './mysql/user'
2020-03-26T15:52:22.559794Z 0 [ERROR] 1 client is using or hasn't closed the table properly
2020-03-26T15:52:22.560212Z 0 [ERROR] mysqld: Table './mysql/db' is marked as crashed and should be repaired
2020-03-26T15:52:22.560376Z 0 [Warning] Checking table:   './mysql/db'
2020-03-26T15:52:22.560393Z 0 [ERROR] 1 client is using or hasn't closed the table properly
2020-03-26T15:52:22.560754Z 0 [Warning] 'user' entry 'root@localhost' ignored in --skip-name-resolve mode.
2020-03-26T15:52:22.560796Z 0 [Warning] 'user' entry 'mysql.sys@localhost' ignored in --skip-name-resolve mode.
2020-03-26T15:52:22.560844Z 0 [Warning] 'db' entry 'performance_schema mysql.session@localhost' ignored in --skip-name-resolve mode.
2020-03-26T15:52:22.560850Z 0 [Warning] 'db' entry 'sys mysql.sys@localhost' ignored in --skip-name-resolve mode.
2020-03-26T15:52:22.560875Z 0 [Warning] 'proxies_priv' entry '@ root@localhost' ignored in --skip-name-resolve mode.
2020-03-26T15:52:22.680437Z 0 [ERROR] mysqld: Table './mysql/tables_priv' is marked as crashed and should be repaired
2020-03-26T15:52:22.680584Z 0 [Warning] Checking table:   './mysql/tables_priv'
2020-03-26T15:52:22.680598Z 0 [ERROR] 1 client is using or hasn't closed the table properly
2020-03-26T15:52:22.681069Z 0 [Warning] 'tables_priv' entry 'user mysql.session@localhost' ignored in --skip-name-resolve mode.
2020-03-26T15:52:22.681089Z 0 [Warning] 'tables_priv' entry 'sys_config mysql.sys@localhost' ignored in --skip-name-resolve mode.
2020-03-26T15:52:22.694107Z 0 [Note] Event Scheduler: Loaded 0 events
```

--- a/Changelog.md
+++ b/Changelog.md
@@ -22,7 +22,7 @@ adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
  * Remove use go module instead of dep
  * Update k8s client to `v0.18.2`
  * Update kubebuilder (to `v2.3.1`) along with controller-runtime (to `v0.6.0`) and controller-gen
- * Update rclone to `v1.52.3`
+ * Update rclone to `v1.53.1`
  * Update `Orchestrator` version from `v3.1.2` to `v3.2.3`
  * Set default MySQL server version to `5.7.31`
 ### Removed

--- a/Changelog.md
+++ b/Changelog.md
@@ -23,11 +23,11 @@ adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
  * Update k8s client to `v0.18.2`
  * Update kubebuilder (to `v2.3.1`) along with controller-runtime (to `v0.6.0`) and controller-gen
  * Update rclone to `v1.52.3`
+ * Update `Orchestrator` version from `v3.1.2` to `v3.2.3`
 ### Removed
 ### Fixed
  * Fix pod labels diff of map
  * Add a lifecycle preStop hook for the  `mysql` container. In `preStop` hook, before killing the master MySQL process, it triggers a `gracefaul-master-takeover-auto` from Orchestrator.
- * Update `Orchestrator` version from `v3.1.2` to `v3.2.2`
  * Fixed backup cleanup job bug (#577)
 
 

--- a/Changelog.md
+++ b/Changelog.md
@@ -24,6 +24,7 @@ adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
  * Update kubebuilder (to `v2.3.1`) along with controller-runtime (to `v0.6.0`) and controller-gen
  * Update rclone to `v1.52.3`
  * Update `Orchestrator` version from `v3.1.2` to `v3.2.3`
+ * Set default MySQL server version to `5.7.31`
 ### Removed
 ### Fixed
  * Fix pod labels diff of map

--- a/Dockerfile.orchestrator
+++ b/Dockerfile.orchestrator
@@ -1,6 +1,7 @@
 ###############################################################################
 #  Build the mysql-oerator related binaries
 ###############################################################################
+
 FROM golang:1.13.14 as builder
 
 # Copy in the go src
@@ -14,34 +15,36 @@ RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o mysql-operator github.c
 RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o mysql-operator-sidecar github.com/presslabs/mysql-operator/cmd/mysql-operator-sidecar
 RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o orc-helper github.com/presslabs/mysql-operator/cmd/orc-helper
 
+
 ##############################################################################
 #  Build Orchestrator
 ###############################################################################
-FROM golang:1.13.7-alpine as builder-orc
+
+FROM golang:1.14.10-alpine3.12 as builder-orc
 
 RUN set -ex \
     && apk add --no-cache \
         bash gcc git musl-dev openssl rsync perl-utils
 
-ARG ORCHESTRATOR_VERSION=v3.1.4
-ARG ORCHESTRATOR_REPO=https://github.com/github/orchestrator.git
+ARG ORCHESTRATOR_VERSION=v3.2.3
+ARG ORCHESTRATOR_REPO=https://github.com/openark/orchestrator.git
 RUN set -ex \
-    && mkdir -p $GOPATH/src/github.com/github/orchestrator \
-    && cd $GOPATH/src/github.com/github/orchestrator \
+    && mkdir -p $GOPATH/src/github.com/openark/orchestrator \
+    && cd $GOPATH/src/github.com/openark/orchestrator \
     && git init && git remote add origin $ORCHESTRATOR_REPO \
     && git fetch --tags \
     && git checkout $ORCHESTRATOR_VERSION
 
-WORKDIR $GOPATH/src/github.com/github/orchestrator
+WORKDIR $GOPATH/src/github.com/openark/orchestrator
 
-RUN set -ex \
-    && ls -l \
-    && ./script/build
+RUN set -ex && ./build.sh -b -P
+
 
 ###############################################################################
 #  Docker image for orchestrator
 ###############################################################################
-FROM alpine:3.11
+
+FROM alpine:3.13
 
 # Create a group and user
 RUN addgroup -g 777 orchestrator && adduser -u 777 -g 777 -S orchestrator
@@ -59,7 +62,7 @@ RUN set -ex \
         tar -C /usr/local/bin -xzv
 
 COPY --chown=777:777 hack/docker/orchestrator/ /
-COPY --from=builder-orc /go/src/github.com/github/orchestrator/bin/ /usr/local/orchestrator/
+COPY --from=builder-orc /tmp/orchestrator-release/build/orchestrator/usr/local/orchestrator/ /usr/local/orchestrator/
 COPY --from=builder /go/src/github.com/presslabs/mysql-operator/orc-helper /usr/local/bin/
 
 USER 777

--- a/Dockerfile.sidecar
+++ b/Dockerfile.sidecar
@@ -1,6 +1,7 @@
 ###############################################################################
 #  Build the mysql-oerator related binaries
 ###############################################################################
+
 FROM golang:1.13.14 as builder
 
 # Copy in the go src
@@ -14,9 +15,11 @@ RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o mysql-operator github.c
 RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o mysql-operator-sidecar github.com/presslabs/mysql-operator/cmd/mysql-operator-sidecar
 RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o orc-helper github.com/presslabs/mysql-operator/cmd/orc-helper
 
+
 ###############################################################################
 #  Build rclone
 ###############################################################################
+
 FROM debian:stretch as rclone
 
 RUN apt-get update \
@@ -36,10 +39,12 @@ RUN wget -nv https://github.com/ncw/rclone/releases/download/v${RCLONE_VERSION}/
     && mv rclone-*-linux-amd64/rclone /usr/local/bin/ \
     && chmod 755 /usr/local/bin/rclone
 
+
 ###############################################################################
 #  Docker image for sidecar containers
 ###############################################################################
-FROM debian:stretch-slim as sidecar
+
+FROM debian:buster-slim as sidecar
 
 RUN groupadd -g 999 mysql
 RUN useradd -u 999 -r -g 999 -s /sbin/nologin \
@@ -51,23 +56,18 @@ RUN apt-get update \
     && rm -rf /var/lib/apt/lists/*
 
 COPY hack/docker/percona.gpg /etc/apt/trusted.gpg.d/percona.gpg
-RUN echo 'deb https://repo.percona.com/apt stretch main' > /etc/apt/sources.list.d/percona.list
+RUN echo 'deb https://repo.percona.com/apt buster main' > /etc/apt/sources.list.d/percona.list
 
 RUN apt-get update \
     && apt-get install -y --no-install-recommends \
-        percona-toolkit percona-xtrabackup-24 unzip \
-    && wget https://github.com/maxbube/mydumper/releases/download/v0.9.5/mydumper_0.9.5-2.stretch_amd64.deb \
-    && dpkg -i mydumper_0.9.5-2.stretch_amd64.deb \
-    && rm -rf mydumper_0.9.5-2.stretch_amd64.deb /var/lib/apt/lists/* \
-    && mkdir -p /usr/share/src \
-    && wget https://github.com/maxbube/mydumper/archive/v0.9.5.tar.gz -O /usr/share/src/mydumper-v0.9.5.tar.gz
+        percona-toolkit percona-xtrabackup-24 unzip default-mysql-client
 
 
 USER mysql
 
-# Copy the mysql-operator-sidecar into its own image
-COPY ./hack/docker/sidecar-entrypoint.sh /usr/local/bin/sidecar-entrypoint.sh
-COPY --from=builder /go/src/github.com/presslabs/mysql-operator/mysql-operator-sidecar /usr/local/bin/mysql-operator-sidecar
 COPY --from=rclone /usr/local/bin/rclone /usr/local/bin/rclone
+# Copy the mysql-operator-sidecar into its own image
+COPY --from=builder /go/src/github.com/presslabs/mysql-operator/mysql-operator-sidecar /usr/local/bin/mysql-operator-sidecar
+COPY ./hack/docker/sidecar-entrypoint.sh /usr/local/bin/sidecar-entrypoint.sh
 
 ENTRYPOINT ["/usr/local/bin/sidecar-entrypoint.sh"]

--- a/Dockerfile.sidecar
+++ b/Dockerfile.sidecar
@@ -29,7 +29,7 @@ RUN apt-get update \
 COPY hack/docker/rclone.gpg /root/rclone.gpg
 RUN gpg --import /root/rclone.gpg
 
-ENV RCLONE_VERSION=1.52.3
+ENV RCLONE_VERSION=1.53.1
 
 RUN wget -nv https://github.com/ncw/rclone/releases/download/v${RCLONE_VERSION}/rclone-v${RCLONE_VERSION}-linux-amd64.zip \
     && wget -nv https://github.com/ncw/rclone/releases/download/v${RCLONE_VERSION}/SHA256SUMS \

--- a/hack/development/Dockerfile.orchestrator
+++ b/hack/development/Dockerfile.orchestrator
@@ -1,27 +1,30 @@
+##############################################################################
+#  Build Orchestrator
 ###############################################################################
-FROM golang:1.13.14-alpine as builder
+FROM golang:1.14.10-alpine3.12 as builder-orc
 
 RUN set -ex \
     && apk add --no-cache \
         bash gcc git musl-dev openssl rsync perl-utils
 
-ARG ORCHESTRATOR_VERSION=v3.2.2
-ARG ORCHESTRATOR_REPO=https://github.com/github/orchestrator.git
+ARG ORCHESTRATOR_VERSION=v3.2.3
+ARG ORCHESTRATOR_REPO=https://github.com/openark/orchestrator.git
 RUN set -ex \
-    && mkdir -p $GOPATH/src/github.com/github/orchestrator \
-    && cd $GOPATH/src/github.com/github/orchestrator \
+    && mkdir -p $GOPATH/src/github.com/openark/orchestrator \
+    && cd $GOPATH/src/github.com/openark/orchestrator \
     && git init && git remote add origin $ORCHESTRATOR_REPO \
     && git fetch --tags \
     && git checkout $ORCHESTRATOR_VERSION
 
-WORKDIR $GOPATH/src/github.com/github/orchestrator
+WORKDIR $GOPATH/src/github.com/openark/orchestrator
 
-RUN set -ex \
-    && ls -l \
-    && ./script/build
+RUN set -ex && ./build.sh -b -P
+
 
 ###############################################################################
-FROM alpine:3.11
+#  Docker image for orchestrator
+###############################################################################
+FROM alpine:3.12
 
 # Create a group and user
 RUN addgroup -g 777 orchestrator && adduser -u 777 -g 777 -S orchestrator
@@ -39,7 +42,7 @@ RUN set -ex \
         tar -C /usr/local/bin -xzv
 
 COPY --chown=777:777 ./hack/docker/orchestrator/ /
-COPY --from=builder /go/src/github.com/github/orchestrator/bin/ /usr/local/orchestrator/
+COPY --from=builder-orc /tmp/orchestrator-release/build/orchestrator/usr/local/orchestrator/ /usr/local/orchestrator/
 COPY ./bin/orc-helper_linux_amd64 /usr/local/bin/orc-helper
 
 USER 777

--- a/hack/development/Dockerfile.sidecar
+++ b/hack/development/Dockerfile.sidecar
@@ -1,6 +1,36 @@
-# Copy the mysql-operator-sidecar into its own image
 # NOTE: this image is for development only
-FROM debian:stretch-slim as sidecar
+# Copy the mysql-operator-sidecar into its own image
+
+
+###############################################################################
+#  Build rclone
+###############################################################################
+
+FROM debian:stretch as rclone
+
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+        gnupg ca-certificates wget unzip
+
+COPY hack/docker/rclone.gpg /root/rclone.gpg
+RUN gpg --import /root/rclone.gpg
+
+ENV RCLONE_VERSION=1.52.3
+
+RUN wget -nv https://github.com/ncw/rclone/releases/download/v${RCLONE_VERSION}/rclone-v${RCLONE_VERSION}-linux-amd64.zip \
+    && wget -nv https://github.com/ncw/rclone/releases/download/v${RCLONE_VERSION}/SHA256SUMS \
+    && gpg --verify --output=- SHA256SUMS > sums \
+    && sha256sum -c --ignore-missing sums \
+    && unzip rclone-*-linux-amd64.zip \
+    && mv rclone-*-linux-amd64/rclone /usr/local/bin/ \
+    && chmod 755 /usr/local/bin/rclone
+
+
+###############################################################################
+#  Docker image for sidecar containers
+###############################################################################
+
+FROM debian:buster-slim as sidecar
 
 RUN groupadd -g 999 mysql
 RUN useradd -u 999 -r -g 999 -s /sbin/nologin \
@@ -9,24 +39,15 @@ RUN useradd -u 999 -r -g 999 -s /sbin/nologin \
 RUN apt-get update \
     && apt-get install -y --no-install-recommends \
         apt-transport-https ca-certificates wget \
-        gnupg1 dirmngr \
     && rm -rf /var/lib/apt/lists/*
 
-RUN apt-key adv --keyserver ha.pool.sks-keyservers.net --recv-keys 9334A25F8507EFA5
-
-RUN echo 'deb https://repo.percona.com/apt stretch main' > /etc/apt/sources.list.d/percona.list
+COPY hack/docker/percona.gpg /etc/apt/trusted.gpg.d/percona.gpg
+RUN echo 'deb https://repo.percona.com/apt buster main' > /etc/apt/sources.list.d/percona.list
 
 RUN apt-get update \
     && apt-get install -y --no-install-recommends \
-        percona-toolkit percona-xtrabackup-24 unzip \
-    && wget https://github.com/maxbube/mydumper/releases/download/v0.9.5/mydumper_0.9.5-2.stretch_amd64.deb \
-    && dpkg -i mydumper_0.9.5-2.stretch_amd64.deb \
-    && rm -rf mydumper_0.9.5-2.stretch_amd64.deb /var/lib/apt/lists/* \
-    && wget https://downloads.rclone.org/rclone-current-linux-amd64.zip \
-    && unzip rclone-current-linux-amd64.zip \
-    && mv rclone-*-linux-amd64/rclone /usr/local/bin/ \
-    && rm -rf rclone-*-linux-amd64 rclone-current-linux-amd64.zip \
-    && chmod 755 /usr/local/bin/rclone
+        percona-toolkit percona-xtrabackup-24 unzip default-mysql-client
+
 
 USER mysql
 
@@ -34,6 +55,8 @@ USER mysql
 # https://support.coreos.com/hc/en-us/articles/115001384693-Tag-Expiration
 LABEL quay.expires-after=2d
 
+COPY --from=rclone /usr/local/bin/rclone /usr/local/bin/rclone
 COPY ./hack/docker/sidecar-entrypoint.sh /usr/local/bin/sidecar-entrypoint.sh
 COPY ./bin/mysql-operator-sidecar_linux_amd64 /usr/local/bin/mysql-operator-sidecar
+
 ENTRYPOINT ["/usr/local/bin/sidecar-entrypoint.sh"]

--- a/hack/development/Dockerfile.sidecar
+++ b/hack/development/Dockerfile.sidecar
@@ -15,7 +15,7 @@ RUN apt-get update \
 COPY hack/docker/rclone.gpg /root/rclone.gpg
 RUN gpg --import /root/rclone.gpg
 
-ENV RCLONE_VERSION=1.52.3
+ENV RCLONE_VERSION=1.53.1
 
 RUN wget -nv https://github.com/ncw/rclone/releases/download/v${RCLONE_VERSION}/rclone-v${RCLONE_VERSION}-linux-amd64.zip \
     && wget -nv https://github.com/ncw/rclone/releases/download/v${RCLONE_VERSION}/SHA256SUMS \

--- a/pkg/controller/mysqlcluster/internal/syncer/config_map.go
+++ b/pkg/controller/mysqlcluster/internal/syncer/config_map.go
@@ -28,13 +28,10 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/intstr"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	logf "sigs.k8s.io/controller-runtime/pkg/log"
 
 	"github.com/presslabs/controller-util/syncer"
 	"github.com/presslabs/mysql-operator/pkg/internal/mysqlcluster"
 )
-
-var log = logf.Log.WithName("config-map-syncer")
 
 // NewConfigMapSyncer returns config map syncer
 func NewConfigMapSyncer(c client.Client, scheme *runtime.Scheme, cluster *mysqlcluster.MysqlCluster) syncer.Interface {

--- a/pkg/controller/mysqlcluster/internal/syncer/config_map.go
+++ b/pkg/controller/mysqlcluster/internal/syncer/config_map.go
@@ -173,7 +173,9 @@ var mysqlMasterSlaveConfigs = map[string]string{
 	"log-bin":           "/var/lib/mysql/mysql-bin",
 	"log-slave-updates": "on",
 
-	"read-only":        "on",
+	// start server without read-only because of https://bugs.mysql.com/bug.php?id=100283
+	// so we can restore from a backup (see https://github.com/presslabs/mysql-operator/issues/509)
+	//"read-only":        "on",
 	"skip-slave-start": "on",
 
 	// Crash safe

--- a/pkg/controller/mysqlcluster/internal/syncer/settings.go
+++ b/pkg/controller/mysqlcluster/internal/syncer/settings.go
@@ -17,9 +17,13 @@ limitations under the License.
 package mysqlcluster
 
 import (
-	"github.com/presslabs/mysql-operator/pkg/util/constants"
 	"k8s.io/apimachinery/pkg/util/intstr"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/presslabs/mysql-operator/pkg/util/constants"
 )
+
+var log = logf.Log.WithName("mysqlcluster-syncer")
 
 // TODO: make those consts private and move them in the file where are used.
 const (

--- a/pkg/controller/mysqlcluster/internal/syncer/statefullset.go
+++ b/pkg/controller/mysqlcluster/internal/syncer/statefullset.go
@@ -510,8 +510,7 @@ func (s *sfsSyncer) ensureVolumes() []core.Volume {
 	} else if s.cluster.Spec.VolumeSpec.EmptyDir != nil {
 		dataVolume.EmptyDir = s.cluster.Spec.VolumeSpec.EmptyDir
 	} else {
-		// warning
-		log.V(-1).Info("no volume spec is specified", "volumeSpec", s.cluster.Spec.VolumeSpec, "key", s.cluster)
+		log.Info("no an allowed volume spec is specified", "volumeSpec", s.cluster.Spec.VolumeSpec, "key", s.cluster)
 	}
 
 	volumes := []core.Volume{

--- a/pkg/sidecar/appconf.go
+++ b/pkg/sidecar/appconf.go
@@ -150,6 +150,10 @@ func initFileQuery(cfg *Config, gtidPurged string) []byte {
 		"SET @@SESSION.SQL_LOG_BIN = 0",
 	}
 
+	// set server as read only
+	// https://github.com/presslabs/mysql-operator/issues/509
+	queries = append(queries, "SET GLOBAL READ_ONLY = 1")
+
 	// create operator database because GRANTS need it
 	queries = append(queries, fmt.Sprintf("CREATE DATABASE IF NOT EXISTS %s", toolsDbName))
 

--- a/pkg/util/constants/constants.go
+++ b/pkg/util/constants/constants.go
@@ -85,7 +85,7 @@ var (
 	MySQLDefaultVersion = semver.MustParse("5.7.31")
 	// MySQLTagsToSemVer maps simple version to semver versions
 	MySQLTagsToSemVer = map[string]string{
-		"5.7": "5.7.26",
+		"5.7": "5.7.31",
 	}
 	// MysqlImageVersions is a map of supported mysql version and their image
 	MysqlImageVersions = map[string]string{

--- a/pkg/util/constants/constants.go
+++ b/pkg/util/constants/constants.go
@@ -82,13 +82,16 @@ const (
 
 var (
 	// MySQLDefaultVersion is the version for mysql that should be used
-	MySQLDefaultVersion = semver.MustParse("5.7.26")
+	MySQLDefaultVersion = semver.MustParse("5.7.31")
 	// MySQLTagsToSemVer maps simple version to semver versions
 	MySQLTagsToSemVer = map[string]string{
 		"5.7": "5.7.26",
 	}
 	// MysqlImageVersions is a map of supported mysql version and their image
 	MysqlImageVersions = map[string]string{
+
+		// percona:5.7.31-centos
+		"5.7.31": "percona@sha256:68dad5e2efeb6893e2d7d116a1eae144f2c641c17d00e7869397395590c91651",
 		// This version of mysql has a bug and doesn't work with the operator,
 		// see: https://github.com/presslabs/mysql-operator/issues/509
 		"5.7.29": "percona@sha256:d801123bbfaf750924f993f5c59189d144a93feb928b8aef95e541dd61c62881",

